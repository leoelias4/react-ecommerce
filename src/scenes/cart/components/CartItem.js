import React from "react";
import PropTypes from "prop-types";
import { css } from "react-emotion";

const styles = {
  root: css`
    display: grid;
    grid-template-columns: 300px auto 50px 50px 160px 100px;
    position: relative;
    grid-gap: 20px;
    min-height: 150px;
    background: #ffffffee;
    padding: 10px;
    box-sizing: border-box;
    align-items: center;
    font-size: 1rem;
    text-align: center;
    margin-bottom: 15px;
    box-shadow: 0px 1px 5px 0px rgba(0, 0, 0, 0.2),
      0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 3px 1px -2px rgba(0, 0, 0, 0.12);
    label: cart-item-root;
    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      grid-gap: 20px;
    }
  `,
  image: css`
    background: #f7f7f7;
    width: 100%;
    height: auto;
    @media (max-width: 768px) {
      height: 160px;
      background: #ddd;
    }
    label: image;
  `,
  removeButton: css`
    outline: none;
    border: none;
    background: transparent;
    cursor: pointer;
    height: 100%;
    border-left: 1px dashed #b7b7b7;
    label: remove-button;
    @media (max-width: 480px) {
      position: absolute;
      width: 45px;
      height: 45px;
      top: 15px;
      right: 15px;
      background: #ffffffa3;
      border-radius: 50%;
      display: flex;
      justify-content: center;
      align-items: center;
      border: none;
    }
  `,
  removeIcon: css`
    width: 30px;
    height: 30px;
    fill: red;
    label: remove-icon;
  `,
  actions: css`
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    label: actions;
  `,
  actionButton: css`
    font-size: 1rem;
    border: 1px solid #d2d2d2;
    border-radius: 2px;
    display: flex;
    justify-content: center;
    align-items: center;
    outline: none;
    cursor: pointer;
    transition: 0.4s all ease;
    label: action-button;
    &:hover {
      background: #f1f1f1;
    }
    &:disabled {
      cursor: not-allowed;
    }
  `
};

const CartItem = props => {
  return (
    <div className={ styles.root }>
      <img className={ styles.image } src={ props.picture } alt={ props.title } />
      <strong>{ props.title }</strong>
      <span>Left: { props.itemsLeft }</span>
      <span>R${ props.price }</span>
      <div className={ styles.actions }>
        <button disabled={ props.quantity <= 1 } className={ styles.actionButton } onClick={ props.decreaseItem }>
          -
        </button>
        <span>No: { props.quantity }</span>
        <button disabled={ props.itemsLeft === 0 } className={ styles.actionButton } onClick={ props.increaseItem }>
          +
        </button>
      </div>
      <button className={ styles.removeButton } onClick={ props.destroyItem }>
        <svg className={ styles.removeIcon } viewBox="0 0 20 20">
          <path d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
          />
        </svg>
      </button>
    </div>
    );
};

export default CartItem;

CartItem.defaultProps = {
  title: "Product",
  itemsLeft: 0,
  price: 0,
  quantity: 0
};

CartItem.propTypes = {
  picture: PropTypes.string,
  title: PropTypes.string.isRequired,
  itemsLeft: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  quantity: PropTypes.number.isRequired,
  decreaseItem: PropTypes.func,
  increaseItem: PropTypes.func,
  destroyItem: PropTypes.func
};
