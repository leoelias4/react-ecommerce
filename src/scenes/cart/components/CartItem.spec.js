import React from "react";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import CartItem from "./CartItem";
configure({
  adapter: new Adapter()
});

const setup = (title, itemsLeft, price, quantity) => {
  const actions = {
    increaseItem: jest.fn(),
    decreaseItem: jest.fn(),
    destroyItem: jest.fn()
  };

  const component = mount(
    <CartItem title={ title } itemsLeft={ itemsLeft } price={ price } quantity={ quantity } {...actions} />
  );
  return {
    component,
    actions,
    price: component.find("span").at(1),
    decreaseButton: component.find("button").at(0),
    increaseButton: component.find("button").at(1),
    destroyButton: component.find("button").at(2)
  };
};

describe("Cart component", () => {
  it("should render successfully", () => {
    const {component} = setup("1");
    expect(component).toMatchSnapshot();
  });

  it("should render cartitem title", () => {
    const {component} = setup("1");
    expect(component.props().title).toEqual("1");
  });

  it("should render cartitem without props", () => {
    const {component} = setup();
    expect(component.props().price).toEqual(0);
    expect(component.props().quantity).toEqual(0);
    expect(component.props().itemsLeft).toEqual(0);
  });

  it("should render price with dollar sign", () => {

    const {price} = setup("title", 1, 12);
    expect(price.text()).toMatch("R$12");
  });

  it("should not call action on button click with 0 quantity", () => {
    const {decreaseButton, actions} = setup("prod1");
    decreaseButton.simulate("click");
    expect(actions.decreaseItem).not.toBeCalled();
  });

  it("should call action on button click with quantity more than 1", () => {
    const {decreaseButton, actions} = setup("prod1", undefined, undefined, 2);
    decreaseButton.simulate("click");
    expect(actions.decreaseItem).toBeCalled();
  });

  it("should not call increase action on button click with 0 item left", () => {
    const {increaseButton, actions} = setup("prod1", 0);
    increaseButton.simulate("click");
    expect(actions.increaseItem).not.toBeCalled();
  });

  it("should call increase action on button click with 1 item left", () => {
    const {increaseButton, actions} = setup("prod1", 1);
    increaseButton.simulate("click");
    expect(actions.increaseItem).toBeCalled();
  });

  it("should call destroy action on button click", () => {
    const {destroyButton, actions} = setup("prod1", 1);
    destroyButton.simulate("click");
    expect(actions.destroyItem).toBeCalled();
  });
});
