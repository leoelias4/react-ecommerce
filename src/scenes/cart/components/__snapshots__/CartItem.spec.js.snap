// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component should render successfully 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <CartItem
    decreaseItem={[Function]}
    destroyItem={[Function]}
    increaseItem={[Function]}
    itemsLeft={0}
    price={0}
    quantity={0}
    title="1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "decreaseItem": [Function],
      "destroyItem": [Function],
      "increaseItem": [Function],
      "itemsLeft": 0,
      "price": 0,
      "quantity": 0,
      "title": "1",
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="css-1xp1ku7-cart-item-root"
>
        <img
                class="css-18cnjk0-image"
                alt="1"
        />
        <strong>
                1
        </strong>
        <span>
                Left: 
                0
        </span>
        <span>
                R$
                0
        </span>
        <div
                class="css-xffwwn-actions"
        >
                <button
                        disabled=""
                        class="css-invcd1-action-button"
                >
                        -
                </button>
                <span>
                        No: 
                        0
                </span>
                <button
                        disabled=""
                        class="css-invcd1-action-button"
                >
                        +
                </button>
        </div>
        <button
                class="css-5csd1k-remove-button"
        >
                <svg
                        class="css-1d6oeul-remove-icon"
                        viewBox="0 0 20 20"
                >
                        <path
                                d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
                        />
                </svg>
        </button>
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="1"
            className="css-18cnjk0-image"
            src={undefined}
/>,
          <strong>
            1
</strong>,
          <span>
            Left: 
            0
</span>,
          <span>
            R$
            0
</span>,
          <div
            className="css-xffwwn-actions"
>
            <button
                        className="css-invcd1-action-button"
                        disabled={true}
                        onClick={[Function]}
            >
                        -
            </button>
            <span>
                        No: 
                        0
            </span>
            <button
                        className="css-invcd1-action-button"
                        disabled={true}
                        onClick={[Function]}
            >
                        +
            </button>
</div>,
          <button
            className="css-5csd1k-remove-button"
            onClick={[Function]}
>
            <svg
                        className="css-1d6oeul-remove-icon"
                        viewBox="0 0 20 20"
            >
                        <path
                                    d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
                        />
            </svg>
</button>,
        ],
        "className": "css-1xp1ku7-cart-item-root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <img
            class="css-18cnjk0-image"
            alt="1"
/>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "1",
            "className": "css-18cnjk0-image",
            "src": undefined,
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "img",
        },
        Object {
          "instance": <strong>
            1
</strong>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1",
          },
          "ref": null,
          "rendered": Array [
            "1",
          ],
          "type": "strong",
        },
        Object {
          "instance": <span>
            Left: 
            0
</span>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Left: ",
              0,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Left: ",
            "0",
          ],
          "type": "span",
        },
        Object {
          "instance": <span>
            R$
            0
</span>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "R$",
              0,
            ],
          },
          "ref": null,
          "rendered": Array [
            "R$",
            "0",
          ],
          "type": "span",
        },
        Object {
          "instance": <div
            class="css-xffwwn-actions"
>
            <button
                        disabled=""
                        class="css-invcd1-action-button"
            >
                        -
            </button>
            <span>
                        No: 
                        0
            </span>
            <button
                        disabled=""
                        class="css-invcd1-action-button"
            >
                        +
            </button>
</div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="css-invcd1-action-button"
                disabled={true}
                onClick={[Function]}
>
                -
</button>,
              <span>
                No: 
                0
</span>,
              <button
                className="css-invcd1-action-button"
                disabled={true}
                onClick={[Function]}
>
                +
</button>,
            ],
            "className": "css-xffwwn-actions",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <button
                disabled=""
                class="css-invcd1-action-button"
>
                -
</button>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "-",
                "className": "css-invcd1-action-button",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "-",
              ],
              "type": "button",
            },
            Object {
              "instance": <span>
                No: 
                0
</span>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "No: ",
                  0,
                ],
              },
              "ref": null,
              "rendered": Array [
                "No: ",
                "0",
              ],
              "type": "span",
            },
            Object {
              "instance": <button
                disabled=""
                class="css-invcd1-action-button"
>
                +
</button>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "+",
                "className": "css-invcd1-action-button",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "+",
              ],
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": <button
            class="css-5csd1k-remove-button"
>
            <svg
                        class="css-1d6oeul-remove-icon"
                        viewBox="0 0 20 20"
            >
                        <path
                                    d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
                        />
            </svg>
</button>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="css-1d6oeul-remove-icon"
              viewBox="0 0 20 20"
>
              <path
                            d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
              />
</svg>,
            "className": "css-5csd1k-remove-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <svg
                class="css-1d6oeul-remove-icon"
                viewBox="0 0 20 20"
>
                <path
                                d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
                />
</svg>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <path
                  d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
/>,
                "className": "css-1d6oeul-remove-icon",
                "viewBox": "0 0 20 20",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <path
                    d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
/>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "d": "M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "path",
                },
              ],
              "type": "svg",
            },
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "decreaseItem": [Function],
        "destroyItem": [Function],
        "increaseItem": [Function],
        "itemsLeft": 0,
        "price": 0,
        "quantity": 0,
        "title": "1",
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="css-1xp1ku7-cart-item-root"
>
          <img
                    class="css-18cnjk0-image"
                    alt="1"
          />
          <strong>
                    1
          </strong>
          <span>
                    Left: 
                    0
          </span>
          <span>
                    R$
                    0
          </span>
          <div
                    class="css-xffwwn-actions"
          >
                    <button
                              disabled=""
                              class="css-invcd1-action-button"
                    >
                              -
                    </button>
                    <span>
                              No: 
                              0
                    </span>
                    <button
                              disabled=""
                              class="css-invcd1-action-button"
                    >
                              +
                    </button>
          </div>
          <button
                    class="css-5csd1k-remove-button"
          >
                    <svg
                              class="css-1d6oeul-remove-icon"
                              viewBox="0 0 20 20"
                    >
                              <path
                                        d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
                              />
                    </svg>
          </button>
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="1"
              className="css-18cnjk0-image"
              src={undefined}
/>,
            <strong>
              1
</strong>,
            <span>
              Left: 
              0
</span>,
            <span>
              R$
              0
</span>,
            <div
              className="css-xffwwn-actions"
>
              <button
                            className="css-invcd1-action-button"
                            disabled={true}
                            onClick={[Function]}
              >
                            -
              </button>
              <span>
                            No: 
                            0
              </span>
              <button
                            className="css-invcd1-action-button"
                            disabled={true}
                            onClick={[Function]}
              >
                            +
              </button>
</div>,
            <button
              className="css-5csd1k-remove-button"
              onClick={[Function]}
>
              <svg
                            className="css-1d6oeul-remove-icon"
                            viewBox="0 0 20 20"
              >
                            <path
                                          d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
                            />
              </svg>
</button>,
          ],
          "className": "css-1xp1ku7-cart-item-root",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <img
              class="css-18cnjk0-image"
              alt="1"
/>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "1",
              "className": "css-18cnjk0-image",
              "src": undefined,
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "img",
          },
          Object {
            "instance": <strong>
              1
</strong>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1",
            },
            "ref": null,
            "rendered": Array [
              "1",
            ],
            "type": "strong",
          },
          Object {
            "instance": <span>
              Left: 
              0
</span>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Left: ",
                0,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Left: ",
              "0",
            ],
            "type": "span",
          },
          Object {
            "instance": <span>
              R$
              0
</span>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "R$",
                0,
              ],
            },
            "ref": null,
            "rendered": Array [
              "R$",
              "0",
            ],
            "type": "span",
          },
          Object {
            "instance": <div
              class="css-xffwwn-actions"
>
              <button
                            disabled=""
                            class="css-invcd1-action-button"
              >
                            -
              </button>
              <span>
                            No: 
                            0
              </span>
              <button
                            disabled=""
                            class="css-invcd1-action-button"
              >
                            +
              </button>
</div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="css-invcd1-action-button"
                  disabled={true}
                  onClick={[Function]}
>
                  -
</button>,
                <span>
                  No: 
                  0
</span>,
                <button
                  className="css-invcd1-action-button"
                  disabled={true}
                  onClick={[Function]}
>
                  +
</button>,
              ],
              "className": "css-xffwwn-actions",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <button
                  disabled=""
                  class="css-invcd1-action-button"
>
                  -
</button>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "-",
                  "className": "css-invcd1-action-button",
                  "disabled": true,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "-",
                ],
                "type": "button",
              },
              Object {
                "instance": <span>
                  No: 
                  0
</span>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "No: ",
                    0,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "No: ",
                  "0",
                ],
                "type": "span",
              },
              Object {
                "instance": <button
                  disabled=""
                  class="css-invcd1-action-button"
>
                  +
</button>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "+",
                  "className": "css-invcd1-action-button",
                  "disabled": true,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "+",
                ],
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": <button
              class="css-5csd1k-remove-button"
>
              <svg
                            class="css-1d6oeul-remove-icon"
                            viewBox="0 0 20 20"
              >
                            <path
                                          d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
                            />
              </svg>
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <svg
                className="css-1d6oeul-remove-icon"
                viewBox="0 0 20 20"
>
                <path
                                d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
                />
</svg>,
              "className": "css-5csd1k-remove-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <svg
                  class="css-1d6oeul-remove-icon"
                  viewBox="0 0 20 20"
>
                  <path
                                    d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
                  />
</svg>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <path
                    d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
/>,
                  "className": "css-1d6oeul-remove-icon",
                  "viewBox": "0 0 20 20",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <path
                      d="M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z"
/>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "d": "M10 8.586l-7.071-7.071-1.414 1.414 7.071 7.071-7.071 7.071 1.414 1.414 7.071-7.071 7.071 7.071 1.414-1.414-7.071-7.071 7.071-7.071-1.414-1.414-7.071 7.071z",
                    },
                    "ref": null,
                    "rendered": Array [
                      undefined,
                    ],
                    "type": "path",
                  },
                ],
                "type": "svg",
              },
            ],
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
